### ISSUE TITLE
HGF: theta vs omega3

### ISSUE THREAD

**Issue by azabicki** [translationalneuromodeling/tapas#170]
Dear Dr Mathys,

first of all, thank you for your excellent papers and detailed documentation. It has already helped me a lot, as well as the discussed issues here in Github.

Recently I have started to work with computational models, as in our group HGF has been used a lot in the last years (I assume you know Simone from her time in London) and I will of course apply it to the data we are currently collecting. So I grabbed the latest version from the Github repo and tried to get it working with a recent dataset. And thanks to the very good documentation and demo files, it works almost right away :)

But since in all papers published from our lab thetas were mentioned and described in the method sections, I was initially irritated to find no more theta in the current version, and that it seems to have been replaced by omega3.  Thus, I got a version of the HGF toolbox from Simone that she has used in the past (it is version 3.0, and from what I have found out in the last few days, the last one with theta implemented). Then I ran `tapas_fitModel([], u, 'tapas_hgf_binary_config', 'tapas_bayes_optimal_binary_config')` in the old as well as in the latest HGF version, keeping identical priors (old: logitthmu = -6, and new: ommu(3) = -6). The aim was to see if the results are similar, or, if not, to find out why and what changed... 

After seeing that the estimates of `theta = 0.0025` and `omega3 = -5.9916` are very different, I did further research (both in the code and online). And fortunately, I found your answer here in another issue ([#78](https://github.com/translationalneuromodeling/tapas/issues/78#issuecomment-638997853)), which explains that

> "theta = exp(omega3). At some point in the history of the HGF Toolbox, we switched from estimating theta in logit space to estimating omega3 in native space. This has the advantage of being more intuitive in that it corresponds to omega2, simply one level higher. "

That was very helpful! So I tried to understand what happens to the priors during estimation and when the values are in "estimation space" and when they are in "native space".

And here (after all the storytelling) is what I don't understand and what I've stumbled upon. From what I understood from the code, this is what happens with `theta` in v3.0:
- the config file sets `theta = -6` in logit space.
- this is estimated and stored in `ptrans(14)=-5.9926`
- to calculate the trajectories, theta is always transformed into native space in the `tapas_hgf_binary_transp.m` function by a sigmoid function (`sgm(-5.9926) ~ 0.0025`)
- and finally this `theta = 0.0025` is reported in native space, e.g. in the Vossel (2014) saccade paper 

while in the latest HGF version this is done with `omega3`:
- the configuration file specifies `omega3 = -6`, already in native space
- the optimised value is then `omega3 = -5.9916 `
- and, to calculate the trajectories in the `tapas_hgf_binary.m` functions, this omega3 is transformed back to theta space in line 46 (`th = exp(p(5 * l-1));`), my explanation for this is to bring omega3 back to the "theta space" where the trajectories are originally calculated ->

What I don't understand is why this is not done by a sigmoid function? If theta was estimated in logit space, and omega3 now in native space, then the transformation function should be a sigmoid, i.e. `theta = sigmoid(omega3)`, right? 

In this case, the difference between theta and exp(omega3) is small because the sigmoid and exp functions are very similar for n<-3, but the difference becomes larger when the value is small. 

Please excuse my perhaps somewhat "pramagmatic" reasoning, as I am not that familiar with the whole underlying Bayesian framework and the theoretical implications or derivations of the mathematical formulae. Which of course can be the reason that I just don't see that everything is alright :)

In summary, I am a bit confused and not sure whether to use the latest HGF version or v3.0? Also to be able to compare my results with older ones from our lab? Also important to know would be if the changes also affect omega2 strongly? As this is a much more interesting parameter for us. 

Thanks in advance for your time! And I really hope I made it understandable what I am struggling with!?

Best regards,
Adam

**Comment by chmathys** [comment]
Dear Adam,

Since omega3 is estimated in native space with a Gaussian prior and omega3 = log(theta), theta has a log-Gaussian prior in the current implementation. In the previous implementation (still present in v3.0), theta had a logit-Gaussian prior and an upper bound.

Because of the different priors, these are two different models despite the simple relation between omega3 and theta. Both are valid. The original implementation has stability advantages because theta is upper-bounded. However, the current implementation is conceptually clearer because of the logical correspondence between the omegas at different levels and the fact that all omegas are estimated in native space.

You can use whichever prior suits your needs better.

Best wishes,
Christoph

Sent with ProtonMail Secure Email.

------- Original Message -------
